x-hooks/use-chain.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+
+import { useRouter } from "next/router";
+
+import { useAppState } from "contexts/app-state";
+
+import { SupportedChainData } from "interfaces/supported-chain-data";
+
+interface FindSupportedChainProps {
+  chainId?: number;
+  chainShortName?: string;
+}
+
+export default function useChain() {
+  const { query } = useRouter();
+  
+  const [chain, setChain] = useState<SupportedChainData>();
+
+  const { state: { supportedChains } } = useAppState();
+
+  function findSupportedChain({ chainId, chainShortName } : FindSupportedChainProps) {
+    return supportedChains?.find(e => e.chainId === chainId || e.chainShortName === chainShortName);
+  }
+
+  function getChainFromUrl() {
+    if (!supportedChains?.length || !query.chain) return undefined;
+    
+    return findSupportedChain({ chainShortName: query.chain.toString() });
+  }
+
+  useEffect(() => {
```

Comment:
This useEffect changes the state globally, therefore _it can't_ be ran inside the `useChain` as this will be called more times than the times we are accounting for.

export `changeChainFromUrl()` that is a `setChain(getChainFromUrl());` and call it on `global-effects`.

Reply:
@moshmage this hook only changes a local state, it was created to group the logic of identifying the chain using the `query.chain` param

Reply:
@vhcsilva I could have _swear_ I read `dispatch()` instead of `setChain` -- My bad.